hugo/content/3-queries/operators.md

```diff
@@ -0,0 +1,324 @@
++++
+next = "/4-updates/"
+prev = "/3-queries/introduction"
+title = "Operators"
+toc = true
+weight = 1
+
++++
+
+The following is a list of the available **query operators** for the Mangrove query builder.
+This page follows the structure of the
+{{% a_blank "MongoDB query operator documentation" "https://docs.mongodb.com/manual/reference/operator/query/" %}}.
+
+### Comparison Operators
+The following operators can be expressed using built-in C++ operators such as `<` and `!=`.
+They work on any field type, and the right-hand value must be the same type as the field.
+Free functions are also provided in the `mangrove` namespace that can be used in lieu of the overloaded
+operators.
+
+- `{{% a_blank "$eq" "https://docs.mongodb.com/manual/reference/operator/query/eq/" %}}`
+ ```cpp
+ // { age: {$eq: 21} }
+ auto results = collection.find(MANGROVE_KEY(User::age) == 21);
+ auto results = collection.find(mangrove::eq(MANGROVE_KEY(User::age), 21));
+ ```
+
+- `{{% a_blank "$gt" "https://docs.mongodb.com/manual/reference/operator/query/gt/" %}}`
+ ```cpp
+ // { age: {$gt: 21} }
+ auto results = collection.find(MANGROVE_KEY(User::age) > 21);
+ auto results = collection.find(mangrove::gt(MANGROVE_KEY(User::age), 21));
+ ```
+
+- `{{% a_blank "$gte" "https://docs.mongodb.com/manual/reference/operator/query/gte/" %}}`
+ ```cpp
+ // { age: {$gte: 21} }
+ auto results = collection.find(MANGROVE_KEY(User::age) >= 21);
+ auto results = collection.find(mangrove::gte(MANGROVE_KEY(User::age), 21));
+ ```
+
+- `{{% a_blank "$lt" "https://docs.mongodb.com/manual/reference/operator/query/lt/" %}}`
+ ```cpp
+ // { age: {$lt: 21} }
+ auto results = collection.find(MANGROVE_KEY(User::age) < 21);
+ auto results = collection.find(mangrove::lt(MANGROVE_KEY(User::age), 21));
+ ```
+
+- `{{% a_blank "$lte" "https://docs.mongodb.com/manual/reference/operator/query/lte/" %}}`
+ ```cpp
+ // { age: {$lte: 21} }
+ auto results = collection.find(MANGROVE_KEY(User::age) <= 21);
+ auto results = collection.find(mangrove::lte(MANGROVE_KEY(User::age), 21));
+ ```
+
+- `{{% a_blank "$ne" "https://docs.mongodb.com/manual/reference/operator/query/ne/" %}}`
+ ```cpp
+ // { age: {$ne: 21} }
+ auto results = collection.find(MANGROVE_KEY(User::age) != 21);
+ auto results = collection.find(mangrove::ne(MANGROVE_KEY(User::age), 21));
+ ```
+
+The following operators are provided as member functions of the name-value pairs.
```

Comment:
I don't think there's a mention of "name-value pairs" before here. It's definitely worth going into a bit of detail, especially because the phrase is repeated throughout.